lib/api/multiObjectDelete.js

```diff
@@ -124,7 +125,7 @@ function _formatXML(quietSetting, errorResults, deleted) {
         if (isDeletingDeleteMarker) {
```

Comment:
[AWS document](http://docs.aws.amazon.com/AmazonS3/latest/API/multiobjectdeleteapi.html) states we should return `DeleteMarkerVersionId` whether creating or deleting a delete marker:

> If the specific delete request in the Multi-Object Delete either creates or deletes a delete marker, Amazon S3 returns this element in response with the version ID of the delete marker. When deleting an object in a bucket with versioning enabled, this value is present for the following two reasons [...]

According to the document, I think we should get rid of `isDeletingDeleteMarker` and stick this xml in the previous check, `if (isDeleteMarker)`. We should also rename `deletingDeleteMarker` to `deleteMarkerVersionId` so it's not confusing. And `isDeleteMarker` should only be true if we are passing `deletingDeleteMarker`/`deleteMarkerVersionId` so we can also replace / get rid of `isDeleteMarker` if we want and check for `if (deleteMarkerVersionId)` instead.

It was probably confusing due to the existing code -- sorry, I must have mis-interpreted the docs while refactoring and wrote this with the wrong interpretation. Please double-check and make sure I'm not wrong again by checking AWS behavior and making tests if we don't already have them.